tv_apps/browser/js/smart_list.js

```diff
@@ -0,0 +1,346 @@
+'use strict';
+
+(function(exports){
+
+	const SMART_LIST_ROOT_FOLDER = '/';
+
+	const SMART_LIST_EVENTS = [
+		'onclick',
+		'onkeyup',
+		'onmouseup',
+		'onmouseout',
+		'onmouseover'
+	];
+
+	function SmartList(options){
+		this.onRender = options.onRender &&
+			typeof options.onRender === 'function' ?
+		 	options.onRender : null;
+
+		this.handleItemKeyUp = options.handleItemKeyUp &&
+			typeof options.handleItemKeyUp === 'function' ?
+		 	options.handleItemKeyUp : null;
+
+		this.handleItemMouseUp = options.handleItemMouseUp &&
+			typeof options.handleItemMouseUp === 'function' ?
+		 	options.handleItemMouseUp : null;
+
+		this.handleListMouseOut = options.handleListMouseOut &&
+			typeof options.handleListMouseOut === 'function' ?
+			options.handleListMouseOut : null;
+
+		this.handleListMouseOver = options.handleListMouseOver &&
+			typeof options.handleListMouseOver === 'function' ?
+			options.handleListMouseOver : null;
+	}
+
+	SmartList.prototype = {
+		init: function(options){
```

Comment:
Missing ```constructor: SmartList,```. Although this doesn't affect the feature, changing prototype like this removes its constructor property.